test/tz_test/validate.cpp

```diff
@@ -90,9 +90,9 @@ tzmain()
     auto& db = get_tzdb();
     std::vector<std::string> names;
     names.reserve(db.zones.size() + db.links.size());
-    for (auto& zone : db.zones)
+    for (const auto& zone : db.zones)
```

Comment:
But I want explicitness because I always want my reference to be const to explicitly show I am not modifying any objects in the loop even if the container would let me. Doing that adds to readability and clarity imho. Conciseness is nothing if it introduces ambiguity. Your comments contradict that idea. Why does that article not convince you?

I also note that the C++ committee rejected an even more "concise" for (e : c) syntax proposal, that was short hand exactly for your for (auto&& e : c) suggestion and I suspect they rejected that proposal exactly for the reason I am suggesting - i.e. people would  be encouraged to use it because it is "concise" but then fail to convey their intent not to modify the container. So I suspect the Committee too decided clarity is more important than being concise here. Have I have convinced you? If not, perhaps you can link something to support your ideas more or explain what it is in the article I linked or that I have said that you find unconvincing?